Domain/Logic/Deserialize/RuntimeFeatureDeserializer.cs

```
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using Microsoft.Extensions.Caching.Memory;
+using Olive;
+using System.Linq;
+using System.Threading.Tasks;
+using Nito.AsyncEx.Synchronous;
+
+namespace Olive.Microservices.Hub
+{
+    public class RuntimeFeature
+    {
+        public string Features;
+        public string Icon;
+        public string Url;
+        public string Permissions;
+        public string Desc;
+    }
+    internal static class RuntimeFeatureDeserializer
+    {
+        static MemoryCache CachedFeatures = new MemoryCache(new MemoryCacheOptions());
+        public static IEnumerable<Feature> GetCachedFeatures()
+        {
+
+            try
+            {
+                using (var stream = new Amazon.S3.Transfer.TransferUtility().OpenStream(Config.GetOrThrow("Authentication:HubFeaturesBucket"), "Features"))
+                {
+                    var task = Task.Run(async () => await stream.ReadAllText());
+                    var result = task.WaitAndUnwrapException();
+                    return Newtonsoft.Json.JsonConvert.DeserializeObject<List<Feature>>(result);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.For(typeof(RuntimeFeatureDeserializer)).Error(ex.ToString());
+                return Enumerable.Empty<Feature>();
+            }
+
+        }
+        public async static Task SetFeaturesFromMicroservice()
+        {
+            var cachedFeatures = new List<Feature>();
+            try
+            {
+                using (var stream = new Amazon.S3.Transfer.TransferUtility().OpenStream(Config.GetOrThrow("Authentication:HubFeaturesBucket"), "Features"))
+                    cachedFeatures = Newtonsoft.Json.JsonConvert.DeserializeObject<List<Feature>>(await stream.ReadAllText());
+
+            }
+            catch (Exception ex)
+            {
+                Log.For(typeof(RuntimeFeatureDeserializer)).Error(ex.ToString());
```

Remove try/catch. Let it fail